sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/implementation/jdbc/JdbcConnectionString.java

```diff
@@ -68,69 +81,128 @@ private void resolveSegments() {
 
-        if (!this.hasProperties()) {
-            builder.append(databaseType.getPathQueryDelimiter());
-        } else {
-            builder.append(databaseType.getQueryDelimiter());
-        }
+        StringBuilder builder = new StringBuilder(this.baseUrl).append(databaseType.getPathQueryDelimiter());
+
+        Map<String, String> mergedProperties = new TreeMap<>(originalProperties);
+        mergedProperties.putAll(this.enhancedProperties);
+
+        this.orderedOriginalPropertyKeys.forEach(k -> builder
+            .append(constructPropertyString(k, mergedProperties.remove(k)))
+            .append(databaseType.getQueryDelimiter()));
+
+        mergedProperties.forEach((k, v) -> builder.append(k).append("=").append(v).append(databaseType.getQueryDelimiter()));
+
+        String enhancedUrl = builder.toString();
+        return enhancedUrl.substring(0, enhancedUrl.length() - 1);
+    }
+
+    private static String constructPropertyString(String key, String value) {
+        return value == null ? key : (key + "=" + value);
+    }
+
+    public void enhanceProperties(Map<String, String> enhancedProperties) {
+        this.enhanceProperties(enhancedProperties, false);
+    }
 
+    public void enhanceProperties(Map<String, String> enhancedProperties,
+                                  boolean silentWhenInconsistentValuePresent) {
         for (Map.Entry<String, String> entry : enhancedProperties.entrySet()) {
             String key = entry.getKey(), value = entry.getValue();
-            String valueProvidedInConnectionString = this.getProperty(key);
+            String valueProvidedInConnectionString = this.originalProperties.get(key);
 
             if (valueProvidedInConnectionString == null) {
-                builder.append(key)
-                    .append("=")
-                    .append(value)
-                    .append(databaseType.getQueryDelimiter());
+                this.enhancedProperties.put(key, value);
             } else if (!value.equals(valueProvidedInConnectionString)) {
-                LOGGER.debug("The property {} is set to another value than default {}", key, value);
-                throw new IllegalArgumentException("Inconsistent property detected");
+                if (silentWhenInconsistentValuePresent) {
+                    LOGGER.debug("The property {} is set to another value than default {}", key, value);
+                } else {
+                    throw new IllegalArgumentException("Inconsistent property of key [" + key +  "] detected");
+                }
             } else {
                 LOGGER.debug("The property {} is already set", key);
             }
         }
-
-        String enhancedUrl = builder.toString();
-        return enhancedUrl.substring(0, enhancedUrl.length() - 1);
     }
 
-    public String getProperty(String key) {
-        return this.properties.get(key);
+    public void enhancePropertyAttributes(String propertyKey, Map<String, String> enhancedAttributes,
```

Comment:
I am a little confused with the two names: `enhancePropertyAttributes` and `enhanceProperties` .

Reply:
Property is something like `useSSL`, `connectionAttributes`, attribute is an attribute in a property, like for MySQL, it can set multiple value to the `connectionAttributes` => `connectionAttributes=_client_name:abc,_client_version:123,_extension_version:cd/123`.